src/components/Expenses.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react';
+import propTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { currencyThunk, addThunk, editSelected, idSelected } from '../actions';
+import './Expenses.css';
+
+class Expenses extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: 0,
+      value: '',
+      description: '',
+      currency: '',
+      method: '',
+      tag: '',
+      exchangeRates: {},
+    };
+
+    this.handleInput = this.handleInput.bind(this);
+    this.saveExpense = this.saveExpense.bind(this);
+    this.getExpenseForEdition = this.getExpenseForEdition.bind(this);
+  }
+
+  componentDidMount() {
+    const { setCurrency } = this.props;
+    setCurrency();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { isEditing } = this.props;
+
+    if (isEditing !== null && isEditing !== prevProps.isEditing) {
+      this.getExpenseForEdition();
+    }
+  }
+
+  getExpenseForEdition() {
+    const { isEditing, expenses } = this.props;
+    const expenseForEdition = expenses.find((expense) => expense.id === isEditing);
+    this.setState({
+      ...expenseForEdition,
+    });
+  }
+
+  handleInput(event) {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
```

Comment:
```suggestion
  handleInput({ target: { name, value} }) {
    this.setState({
      [name]: value,
    });
  }
```

90% de certeza que isso funciona! 🤔